static/app/views/profiling/content.tsx

```diff
@@ -28,7 +33,22 @@ interface ProfilingContentProps {
 function ProfilingContent({location, selection}: ProfilingContentProps) {
   const organization = useOrganization();
   const cursor = decodeScalar(location.query.cursor);
-  const [requestState, traces, pageLinks] = useProfiles({cursor, selection});
+  const profileFilters = useProfileFilters();
+  const query = decodeScalar(location.query.query);
+  const [requestState, traces, pageLinks] = useProfiles({cursor, query, selection});
+
+  const handleSearch = (searchQuery: string) => {
+    browserHistory.push({
+      ...location,
+      query: {
+        ...location.query,
+        cursor: undefined,
+        query: searchQuery.trim() || undefined,
+      },
+    });
+  };
+
+  const searchQuery = decodeScalar(location.query.query, '').trim();
```

Comment:
You are already decoding the query on L37. iirc trim creates a copy so you can just query.trim()

Why do we even have to trim this btw? I always hate when inputs trim the value, especially painful when you are sometimes new to the product. If we can, I think we should just leave the query be :D

Reply:
i think this was just me copying existing work, but the `.trim()` here really isn't needed